starlite/openapi/utils.py

```diff
@@ -31,3 +35,30 @@ def get_openapi_type_for_complex_type(field: "SignatureField") -> "OpenAPIType":
         "not be documented as a parameter, annotate it using the `Dependency` function, e.g., "
         f"`{field.name}: ... = Dependency(...)`."
     )
+
+
+def default_operation_id_creator(
+    route_handler: "HTTPRouteHandler",
+    http_method: "Method",
+    path_components: List[Union[str, "PathParameterDefinition"]],
+) -> str:
+    """Create a unique 'operationId' for an OpenAPI PathItem entry.
+
+    :param route_handler: The HTTP Route Handler instance.
+    :param http_method: The HTTP method for the given PathItem.
+    :param path_components: A list of path components.
+    :return: A camelCased operationId created from the handler function name, http method and path components.
+    """
+
+    handler_namespace = (
+        http_method.title() + route_handler.handler_name.title()
+        if len(route_handler.http_methods) > 1
+        else route_handler.handler_name.title()
+    )
+
+    components_namespace = ""
+    for component in [c if not isinstance(c, PathParameterDefinition) else c.name for c in path_components]:
```

Comment:
this is only a very minor nit, but a generator suffices here - the list doesn't need to be constructed to iterate over the contents.

```suggestion
    for component in (c if not isinstance(c, PathParameterDefinition) else c.name for c in path_components):
```